src/assets/components/ProductList.tsx

```diff
@@ -0,0 +1,212 @@
+      regex.test(part) ? <strong key={index} className="text-yellow-500">{part}</strong> : part
+    );
+  };
+
+  // Apply filters to products
+  useEffect(() => {
+    const applyFilters = () => {
+      const filtered = products.filter((product) => {
+        const searchLower = searchQuery.toLowerCase();
+        const matchesSearch =
+          product.name.toLowerCase().includes(searchLower) ||
+          product.description.toLowerCase().includes(searchLower) ||
+          product.categories.some((cat) => cat.toLowerCase().includes(searchLower)) ||
+          product.brand.toLowerCase().includes(searchLower);
+
+        const matchesCategoryLvl0 = filters.categoryLvl0
+          ? product.categories.includes(filters.categoryLvl0)
+          : true;
+
+        const matchesCategoryLvl1 = filters.categoryLvl1
+          ? product.categories.includes(filters.categoryLvl1)
+          : true;
+
+        const matchesBrand = filters.brand
+          ? product.brand.toLowerCase() === filters.brand.toLowerCase()
+          : true;
+
+        const matchesPrice =
+          product.price >= filters.priceRange[0] && product.price <= filters.priceRange[1];
+
+        const matchesFreeShipping = filters.freeShipping ? product.free_shipping : true;
+
+        const matchesRating = filters.rating ? product.rating >= filters.rating : true;
+
+        return (
+          matchesSearch &&
+          matchesCategoryLvl0 &&
+          matchesCategoryLvl1 &&
+          matchesBrand &&
+          matchesPrice &&
+          matchesFreeShipping &&
+          matchesRating
+        );
+      });
+
+      setFilteredProducts(filtered);
+    };
+
+    applyFilters();
+  }, [filters, products, searchQuery]);
+
+  // Update URL when filters change
+  useEffect(() => {
+    const params = new URLSearchParams();
+    if (filters.categoryLvl0) params.set('categoryLvl0', filters.categoryLvl0);
+    if (filters.categoryLvl1) params.set('categoryLvl1', filters.categoryLvl1);
+    if (filters.brand) params.set('brand', filters.brand);
+    if (filters.priceRange[0] !== 84) params.set('priceMin', filters.priceRange[0].toString());
+    if (filters.priceRange[1] !== 4000) params.set('priceMax', filters.priceRange[1].toString());
+    if (filters.freeShipping) params.set('freeShipping', filters.freeShipping.toString());
+    if (filters.rating !== 0) params.set('rating', filters.rating.toString());
+
+    navigate({ search: params.toString() }, { replace: true });
+  }, [filters, navigate]);
+
+  const handleFilterChange = (newFilters: typeof filters) => {
+    setFilters(newFilters);
+  };
+
+  const clearFilters = () => {
+    setFilters({
+      categoryLvl0: '',
+      categoryLvl1: '',
+      brand: '',
+      priceRange: [84, 4000],
+      freeShipping: false,
+      rating: 0,
+    });
+    setFilterKey((prevKey) => prevKey + 1);
+    navigate({ search: '' }, { replace: true });
+  };
+
+  return (
+    <div className="flex container">
```

Comment:
```suggestion
    <div className="flex container mx-auto">
```
Căn giữa giao diện, ở các màn hình to đang bị lệch về bên trái